editor/src/components/editor/project-fork-flow.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { useTriggerForkProject } from './persistence-hooks'
+import { Substores, useEditorState } from './store/store-hook'
+
+export const ForkSearchParamKey = 'fork'
+
+export const ProjectForkFlow = React.memo(() => {
+  const searchParams = new URLSearchParams(window.location.search)
+  const shouldFork = searchParams.get(ForkSearchParamKey) === 'true'
+
+  const projectName = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.projectName,
+    'ProjectForkFlow projectName',
+  )
+
+  const triggerForkProject = useTriggerForkProject()
+
+  React.useEffect(() => {
+    if (shouldFork && projectName != null) {
```

Comment:
shouldn't this check projectId as opposed to projectName?

Reply:
that's right! fixed in [`4ebb10c` (#4910)](https://github.com/concrete-utopia/utopia/pull/4910/commits/4ebb10c43d117448fc409979dee1afa96c9fb9d0)